src/main/java/org/openmaptiles/layers/Water.java

```diff
@@ -121,11 +175,123 @@ public void process(Tables.OsmWaterPolygon element, FeatureCollector features) {
         .setAttr(Fields.INTERMITTENT, element.isIntermittent() ? 1 : 0)
         .setAttrWithMinzoom(Fields.BRUNNEL, Utils.brunnel(element.isBridge(), element.isTunnel()), 12)
         .setAttr(Fields.CLASS, clazz);
+
+      try {
+        var geom = element.source().worldGeometry();
+        try {
+          attemptNeLakeIdMapping(element, geom);
+        } catch (TopologyException e) {
+          try {
+            var fixedGeom = GeometryFixer.fix(geom);
+            attemptNeLakeIdMapping(element, fixedGeom);
+          } catch (TopologyException e2) {
+            throw new GeometryException("fix_omt_water_topology_error",
+              "error fixing polygon: " + e2 + "; original error: " + e);
```

Comment:
I see this is firing 16 times when run over the planet, but I can't tell which line is actually causing this failure?  Can you catch the TopologyException closer to where it happens so that 1) we get a more detailed error message that helps pinpoint where the issue happened and 2) we can skip the outer layer of try/catch(geometry exception) and just have the final attempt log the error like this: https://github.com/onthegomap/planetiler/blob/12558b99a8074781e2adc9a2130cab5f481b4395/planetiler-core/src/main/java/com/onthegomap/planetiler/geo/GeoUtils.java#L311-L338.  Also please pass `config.logJtsExceptions()` into the 4th argument of `e.log(...)` below which will print out more details about when the error happens?

I understand this is a best-effort attempt to add OSM IDs to low zoom lakes so it's fine if it misses some, we just try to track down any of these robustness issues with JTS pretty aggressively to report to its maintainer.

Here's an example failure for reference:

> 0:56:26 WAR [osm_pass2:process] - Unable to process intersections for OSM feature 13070460: error fixing polygon: org.locationtech.jts.geom.TopologyException: found non-noded intersection between LINESTRING ( 0.7828271086875376 0.3132446324915123, 0.7828058490561336 0.3132069744583109 ) and LINESTRING ( 0.7828058490561336 0.3132069744583109, 0.7828042136998719 0.3132040773818683 ) [(0.7828058490561336, 0.3132069744583109, NaN) ]; original error: org.locationtech.jts.geom.TopologyException: found non-noded intersection between LINESTRING ( 0.7828271086875376 0.3132446324915123, 0.7828058490561336 0.3132069744583109 ) and LINESTRING ( 0.7828058490561336 0.3132069744583109, 0.7828042136998719 0.3132040773818683 ) [ (0.7828058490561336, 0.3132069744583109, NaN) ]